pkg/tasks/encryption_providers.go

```diff
@@ -0,0 +1,191 @@
+	return s.RunTaskOnControlPlane(pushEncryptionConfigurationOnNode, state.RunParallel)
+}
+
+func UploadEncryptionConfigurationWithNewKey(s *state.State) error {
+	s.Logger.Infof("Uploading EncryptionProviders configuration file...")
+
+	if s.LiveCluster.EncryptionConfiguration == nil ||
+		s.LiveCluster.EncryptionConfiguration.Config == nil {
+		return errors.New("failed to read live cluster encryption providers configuration")
+	}
+
+	// oldConfig := s.LiveCluster.EncryptionConfiguration.Config.DeepCopy()
+	if err := encryptionproviders.UpdateEncryptionConfigWithNewKey(s.LiveCluster.EncryptionConfiguration.Config); err != nil {
+		return err
+	}
+
+	config, err := templates.KubernetesToYAML([]runtime.Object{s.LiveCluster.EncryptionConfiguration.Config})
+	if err != nil {
+		return err
+	}
+
+	s.Configuration.AddFile("cfg/encryption-providers.yaml", config)
+	return s.RunTaskOnControlPlane(pushEncryptionConfigurationOnNode, state.RunParallel)
+}
+
+func UploadEncryptionConfigurationWithoutOldKey(s *state.State) error {
+	s.Logger.Infof("Uploading EncryptionProviders configuration file...")
+
+	if s.LiveCluster.EncryptionConfiguration == nil ||
+		s.LiveCluster.EncryptionConfiguration.Config == nil {
+		return errors.New("failed to read live cluster encryption providers configuration")
+	}
+
+	// oldConfig := s.LiveCluster.EncryptionConfiguration.Config.DeepCopy()
+
+	encryptionproviders.UpdateEncryptionConfigRemoveOldKey(s.LiveCluster.EncryptionConfiguration.Config)
+
+	config, err := templates.KubernetesToYAML([]runtime.Object{s.LiveCluster.EncryptionConfiguration.Config})
+	if err != nil {
+		return err
+	}
+	s.Configuration.AddFile("cfg/encryption-providers.yaml", config)
+	return s.RunTaskOnControlPlane(pushEncryptionConfigurationOnNode, state.RunParallel)
+}
+
+func pushEncryptionConfigurationOnNode(s *state.State, node *kubeoneapi.HostConfig, conn ssh.Connection) error {
+	err := s.Configuration.UploadTo(conn, s.WorkDir)
+	if err != nil {
+		return err
+	}
+	cmd, err := scripts.SaveEncryptionProvidersConfig(s.WorkDir, s.GetEncryptionProviderConfigName())
+	if err != nil {
+		return err
+	}
+
+	_, _, err = s.Runner.RunRaw(cmd)
+	return err
+}
+
+func RewriteClusterSecrets(s *state.State) error {
+	s.Logger.Infof("Rewriting cluster secrets...")
+	secrets := corev1.SecretList{}
+	err := s.DynamicClient.List(context.Background(), &secrets, &dynclient.ListOptions{})
+	if err != nil {
+		return err
+	}
+	for i := range secrets.Items {
+		secret := secrets.Items[i]
+		if err = s.DynamicClient.Update(context.Background(), &secret, &dynclient.UpdateOptions{}); err != nil {
+			if kerrors.IsConflict(err) {
+				err = s.DynamicClient.Get(context.Background(), types.NamespacedName{Name: secret.Name, Namespace: secret.Namespace}, &secret)
+				if err != nil {
+					return err
+				}
+				if err = s.DynamicClient.Update(context.Background(), &secret, &dynclient.UpdateOptions{}); err != nil {
+					return err
+				}
+			}
+			return err
+		}
+	}
```

Comment:
We should use `RetryOnConflict` here, like this:
https://github.com/kubermatic/kubeone/blob/eb97eae6cd11f6b723a93bc24c8b4df7868e350b/pkg/tasks/nodes.go#L46-L55